Orange/base.py

```diff
@@ -22,14 +62,15 @@ class Learner:
     learner_adequacy_err_msg = ''
 
     def __init__(self, preprocessors=None):
-        if preprocessors is None:
-            preprocessors = type(self).preprocessors
-        self.preprocessors = list(preprocessors)
+        self.use_default_preprocessors = False
+        if isinstance(preprocessors, Iterable):
+            self.preprocessors = preprocessors
```

Comment:
No, of course, we just want to support iterables. But generators are iterables, and the user might efficiently do:
```py
all_preprocessors = [all the preprocessors Orange supports]
valid = (p for p in all_preprocessors if requires(SomeLearner, p))
learner = SomeLearner(preprocessors=valid)
```
He might obtain those preprocessors otherwise. In any case, we currenlty require the input (if a collection of preprocessors) to be an [Iterable, not a Sequence](https://docs.python.org/3/library/collections.abc.html).
And with above user's code, this happens:
```py
>>> list(learner.active_preprocessors) == list(p for p in all_preprocessors
...                                            if requires(SomeLearner, p))
True
>>> list(learner.active_preprocessors) == list(p for p in all_preprocessors
...                                            if requires(SomeLearner, p))
False
>>> list(learner.active_preprocessors)  # Wat happen??
[]
```
Just wrapping that input, potentially a generator iterable, into tuple avoids this.